integrations/adobe-analytics/lib/heartbeat.js

```diff
@@ -0,0 +1,180 @@
+
+  },
+  chromecastPlaybackExited: function (track) {
+
+    window.ADBMobile.media.trackPause();
+    window.ADBMobile.media.trackSessionEnd();
+  },
+  chromecastVideoPaused: function (track) {
+    window.ADBMobile.media.trackPause();
+
+  },
+  chromecastVideoStart: function (track) {
+    window.ADBMobile.media.trackPlay();
+  },
+  chromecastVideoComplete: function (track) {
+    var mediaMetadata = window.extractMediaMetadata(track); 
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.ChapterComplete, null, mediaMetadata);
+    window.ADBMobile.media.trackComplete();
+  },
+  chromecastSessionEnd: function (track) {
+    window.ADBMobile.media.trackSessionEnd();
+
+  },
+  chromecastAdStarted: function (track) {
+    var props = track.properties();
+
+    var info = {
+      name: props.title || 'no title',
+      id: props.asset_id.toString() || 'default ad',
+      position: props.position || 1,
+      length: props.total_length || 0
+    }
+    var adInfo = window.ADBMobile.media.createAdObject(info.name, info.id, info.position, info.length);
+
+    var standardAdMetadata = {};
+    standardAdMetadata[ADBMobile.media.AdMetadataKeys.ADVERTISER] = props.video_ad_advertiser || null;
+    standardAdMetadata[ADBMobile.media.AdMetadataKeys.CAMPAIGN_ID] = props.video_ad_campaign_id || null;
+
+    if (adInfo) {
+      adInfo[ADBMobile.media.MediaObjectKey.StandardAdMetadata] = standardAdMetadata;
+    }
+
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.AdBreakStart);
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.AdStart, adInfo);
+
+  },
+  chromecastAdCompleted: function (track) {
+    var mediaMetadata = window.extractMediaMetadata(track); 
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.AdComplete, null, mediaMetadata);
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.AdBreakComplete, null, mediaMetadata);
+
+  },
+  chromecastAdSkipped: function (track) {
+    var mediaMetadata = window.extractMediaMetadata(track); 
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.AdSkip, null, mediaMetadata);
+  },
+  chromecastSeekStarted: function (track) {
+    var mediaMetadata = window.extractMediaMetadata(track); 
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.SeekStart, null, mediaMetadata);
+  },
+  chromecastSeekCompleted: function (track) {
+    var mediaMetadata = window.extractMediaMetadata(track); 
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.SeekComplete, null, mediaMetadata);
+  },
+  chromecastBufferStarted: function (track) {
+    var mediaMetadata = window.extractMediaMetadata(track); 
+    window.ADBMobile.media.trackEvent(window.ADBMobile.media.Event.BufferStart, null, mediaMetadata);
+  },
+  chromecastQualityUpdated: function (track) {
+    var mediaMetadata = window.extractMediaMetadata(track); 
+    var props = track.properties();
+    let qosInfo = {
```

Comment:
```suggestion
    var qosInfo = {
```
I believe we cannot use `let` yet in a.js 1.0.